plugin/gc-app/src/main/java/io/deephaven/app/GcPoolsPublisher.java

```diff
@@ -71,28 +70,43 @@ public static Table stats(Table pools) {
                         "Name");
     }
 
-    private final StreamConsumer consumer;
     private WritableChunk<Values>[] chunks;
+    private StreamConsumer consumer;
     private boolean isFirst;
 
-    GcPoolsConsumer(StreamConsumer consumer) {
-        this.consumer = Objects.requireNonNull(consumer);
+    GcPoolsPublisher() {
         // noinspection unchecked
         chunks = StreamToTableAdapter.makeChunksForDefinition(DEFINITION, CHUNK_SIZE);
         isFirst = true;
     }
 
+    @Override
+    public void register(@NotNull StreamConsumer consumer) {
+        if (this.consumer != null) {
+            throw new IllegalStateException("Can not register multiple StreamConsumers.");
+        }
+        this.consumer = Objects.requireNonNull(consumer);
+    }
+
     public synchronized void add(GcInfo info) {
         final Map<String, MemoryUsage> afterMap = info.getMemoryUsageAfterGc();
+        int poolSize = afterMap.size();
+        if (chunks[0].size() + poolSize >= CHUNK_SIZE) {
```

Comment:
I think you need to consider changing your chunk size at runtime if the pool size exceeds the default chunk size.